### ISSUE TITLE
Custom done.txt

### ISSUE THREAD

**Issue by harshad1** [mpcjanssen/simpletask-android#197]
Similar to how it is possible to use a custom file as 'todo.txt' would it be possible to specify the file to be used as 'done.txt' 

The mechanism used to specify 'todo.txt' is already in place. This should make implementing this request simpler.

Thanks :)

**Comment by mpcjanssen** [comment]
Most of the functionality is indeed there. I am just not sure how to show this in the ui. Should it be an application wide setting? Or be shown as part of the archive menu item? Not sure on those yet.

**Comment by harshad1** [comment]
Personally, I think selecting both the todo.txt and done.txt files 
should be done in the settings. Perhaps a section called files and 
archival.

Files and archival could consist of:
-> Select todo.txt
-> Select done.txt
-> Enable auto archival
-> Archive now

My two cents.

**Comment by vojtechkral** [comment]
Please don't remove _Archive now_ from the context menu. I use it regularly...

**Comment by mpcjanssen** [comment]
I wont remove anything just like that (except maybe experimental features)

**Comment by mpcjanssen** [comment]
could possibly be implemented by Lua 
```
function archiveName(todoTxtName)
   return archive fileName 
end
```
Slashed wont be allowed so the archive file name should be in the same folder. This does make it easy to have for example daily archives by smart use of `os.time`

**Comment by smichel17** [comment]
I don't think this should be done in lua. I think it should be done in the gui, at the same time that we think about an interface that allows for easy switching between todo.txt files not in the same folder, like #411.

**Comment by mpcjanssen** [comment]
@smichel I doubt it is possible to make something in the gui that fits all. Doing it in Lua allows several scenarios

- daily done.txt
- done.txt per todo file
- etc

I am not going to spend time making a UI for this. Users have done.txt. if they need anything else they can build it themselves.

**Comment by smichel17** [comment]
I don't think it would require that much gui work. It would almost certainly happen in the settings, and you could re-use the current file switcher as a file picker.

Let's jump into the future and say we went with my idea of moving filter config into the settings. One possibility is that each filter could specify a todo.txt and a done.txt file to be used. Then, you would no longer switch between files using the file switcher like right now; you'd just switch to a filter that uses a different file. 

Apparently I completely overlooked you mentioning daily archives the first time; I hadn't thought of it, and I agree that should be left to lua.

Regardless, imo implementing either the archiveName callback or a gui for file-linked-filters should wait until after I've finished my redesign and you've decided whether you're willing to implement it.

**Comment by smichel17** [comment]
Duplicate of #146
